### PR TITLE
util/log: extend `logpb.Entry` with more fine grained fields

### PR DESCRIPTION
**Pull request by knz**
Needed for #65633 and #61378.

cc @cockroachdb/server

### CODE REVIEW

File: pkg/util/log/format_crdb_v1_test.go
Comment (cameronnunez) [inline pkg/util/log/format_crdb_v1_test.go:122]: Each instance of the string "clog_test.go" in `format_crdb_v1_test.go` and `testdata/crdb_v1` should be changed to the string "format_crdb_v1_test.go"

**Review by cameronnunez — CHANGES_REQUESTED** [review]
After these minor changes, LGTM! Thanks for your work here.

**Review by knz — COMMENTED** [review]
nit: if you are only requesting minor changes, then it's better to leave your review disposition as "neutral" instead of "changes requested".
With "changes requested" I have to wait until you complete the review before I can merge the PR.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66096#-:-MbQiHzu7TfrTAatJXAx:b-vbmgq6)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @cameronnunez)

___
*[pkg/util/log/format_crdb_v1_test.go, line 32 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66096#-MbNLNYoUDQGWSMWUy4l-r3-32:-MbQhqDyE89Ucfj_1qfF:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8cc0921781bafc52a555c48404759e4ad6e3654a/pkg/util/log/format_crdb_v1_test.go#L32)):*
<details><summary><i>Previously, cameronnunez (Cameron Nunez) wrote…</i></summary><blockquote>

```suggestion
func TestFormatCrdbV1Decode(t *testing.T) {
```

For consistency, I suggest a slight change to the function name here.
</blockquote></details>

Done.
___
*[pkg/util/log/format_crdb_v1_test.go, line 122 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66096#-MbNLNYoUDQGWSMWUy4l-r3-122:-MbQhqyx-RO3JocBQWfC:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8cc0921781bafc52a555c48404759e4ad6e3654a/pkg/util/log/format_crdb_v1_test.go#L122)):*
<details><summary><i>Previously, cameronnunez (Cameron Nunez) wrote…</i></summary><blockquote>

Each instance of the string "clog_test.go" in `format_crdb_v1_test.go` and `testdata/crdb_v1` should be changed to the string "format_crdb_v1_test.go"
</blockquote></details>

Done.

**Comment by cameronnunez** [comment]
> nit: if you are only requesting minor changes, then it's better to leave your review disposition as "neutral" instead of "changes requested".
> With "changes requested" I have to wait until you complete the review before I can merge the PR.
Understood!

**Comment by knz** [comment]
You still need to review and change your disposition.
image

**Review by cameronnunez — APPROVED** [review]

**Comment by cameronnunez** [comment]
> You still need to review and change your disposition.
> image

👍 Done.

**Comment by knz** [comment]
awesome, thank you!

bors r=cameronnunez

**Pull request merged**
